### PR TITLE
Add .github directory with actions to build/push configs

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

This PR introduces GitHub actions to tag and build/push configurations. 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).

### How has this code been tested

Tested it in the main branch of [my fork](https://github.com/turkenh/platform-ref-cloud-native):
- Each commit builds and creates a new tag
- Running `ci` action after a new tag builds and pushes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+package.xpkg
```
Comment (hasheddan) [inline .gitignore:1]: nitpick: likely don't ever want to commit an xpkg
```suggestion
*.xpkg
```

**Review by hasheddan — APPROVED** [review]
One nitpick, but LGTM!

**Comment by turkenh** [comment]
@jbw976 we will need to add the two GITHUB secrets here and to other platform-ref repos. 

Do you want to add them or prefer to give me access?

**Comment by jbw976** [comment]
@turkenh i've given you access to this repo to add the secrets and test everything end to end.  Once things are looking good here, then we can repeat for the other platform-ref-* repos :)

**Comment by turkenh** [comment]
@jbw976 just merged and verified that things looking good.

Basically, our release process would be:

1. Run TAG workflow
1. Run CI workflow

Does this sound good to you? If yes, I will just replicate same to other platform-ref repositories.

Screen Shot 2021-06-25 at 00 08 18

**Comment by jbw976** [comment]
Very cool @turkenh!! Thank you for adding this automation and giving us a real release pattern for the reference platforms 💪 

Couple questions (and apologies for maybe going a bit off-topic of this PR):

1. Do you think we should add some explicit documentation for the release process/flow, OR just point to the Crossplane release process since it's similar, or what do you think?
2. I assume tagging the next pre-release should be part of the flow too, e.g. https://crossplane.io/docs/v1.2/contributing/release-process.html#tag-next-pre-release.  I know in the past we've had issues though if you tag the same commit with 2 different tags, so we've done "empty commits" to move master onwards.  I don't see that done for the last provider-aws:v0.19 release though, both the v0.19.0 and v0.20.0-rc.0 tags point to the same commit: https://github.com/crossplane/provider-aws/tags.  Do we still have this problem? 

/cc @hasheddan

**Comment by turkenh** [comment]
Thank you @jbw976!

> 1. Do you think we should add some explicit documentation for the release process/flow, OR just point to the Crossplane release process since it's similar, or what do you think?

My steps above would work if we want to continue as before; tag and release from master. 
But with the Github actions introduced, we can just follow the same release process as Crossplane if we want to do branching in these platform-ref repositories. However, I have some doubt on whether we really need branching here given there is not much active development and whether it pays for the overhead it is adding.

> 2. I assume tagging the next pre-release should be part of the flow too, e.g. https://crossplane.io/docs/v1.2/contributing/release-process.html#tag-next-pre-release. I know in the past we've had issues though if you tag the same commit with 2 different tags, so we've done "empty commits" to move master onwards. I don't see that done for the last provider-aws:v0.19 release though, both the v0.19.0 and v0.20.0-rc.0 tags point to the same commit: https://github.com/crossplane/provider-aws/tags. Do we still have this problem?

I don't have a lot of context about this, but I would expect having different commits to be required for different tags.

**Pull request merged**